bundles/plugins/org.bonitasoft.studio.properties/src-test/java/org/bonitasoft/studio/properties/sections/subprocess/CallActivityHelperTest.java

```diff
@@ -0,0 +1,321 @@
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.bonitasoft.studio.properties.sections.subprocess;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.doReturn;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.bonitasoft.studio.common.emf.tools.ExpressionHelper;
+import org.bonitasoft.studio.common.emf.tools.ModelHelper;
+import org.bonitasoft.studio.common.repository.RepositoryAccessor;
+import org.bonitasoft.studio.diagram.custom.repository.DiagramRepositoryStore;
+import org.bonitasoft.studio.model.expression.builders.ExpressionBuilder;
+import org.bonitasoft.studio.model.process.AbstractProcess;
+import org.bonitasoft.studio.model.process.CallActivity;
+import org.bonitasoft.studio.model.process.ContractInputType;
+import org.bonitasoft.studio.model.process.Data;
+import org.bonitasoft.studio.model.process.InputMapping;
+import org.bonitasoft.studio.model.process.InputMappingAssignationType;
+import org.bonitasoft.studio.model.process.OutputMapping;
+import org.bonitasoft.studio.model.process.Pool;
+import org.bonitasoft.studio.model.process.builders.CallActivityBuilder;
+import org.bonitasoft.studio.model.process.builders.ContractBuilder;
+import org.bonitasoft.studio.model.process.builders.ContractInputBuilder;
+import org.bonitasoft.studio.model.process.builders.DataBuilder;
+import org.bonitasoft.studio.model.process.builders.InputMappingBuilder;
+import org.bonitasoft.studio.model.process.builders.JavaObjectDataBuilder;
+import org.bonitasoft.studio.model.process.builders.OutputMappingBuilder;
+import org.bonitasoft.studio.model.process.builders.PoolBuilder;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CallActivityHelperTest {
+
+    @Mock
+    private RepositoryAccessor repositoryAccessor;
+    @InjectMocks
+    private CallActivityHelper callActivityHelper;
+
+    @Mock
+    DiagramRepositoryStore drs;
+
+    @Before
+    public void setUp() throws Exception {
+        doReturn(drs).when(repositoryAccessor).getRepositoryStore(DiagramRepositoryStore.class);
+    }
+
+    @Test
+    public void testAutoMap_ReturnEmptyWhenNoPoolConfigured() {
+        final CallActivity callActivity = CallActivityBuilder
+                .aCallActivity()
+                .build();
+        final CallActivityMapper mapper = callActivityHelper.automapSubProcess(callActivity);
+        assertThat(mapper.getInputMappingToCreate()).isEmpty();
+        assertThat(mapper.getOutputMappingToCreate()).isEmpty();
+    }
+
+    @Test
+    public void testAutoMap_simple() {
+        final Data callActivityData = DataBuilder.aData().havingDataType(ModelHelper.createStringDataType()).withName("name1").build();
```

Comment:
Nope, I don't want, because the name of the DataType is not initialized with the builder, in real world, we are using the ModelHelper method